### ISSUE TITLE
right click on (image)Button

### ISSUE THREAD

**Issue by keszegh** [ocornut/imgui#4241]
I have searched the code etc. but still could not figure out how to handle right click on an imageButton (just like it behaves with left click) without reinventing the wheel. what is the simplest way?
thanks

**Comment by ocornut** [comment]
What do you intend to do with the “right click” ?

The definition of “right click” may not be what you think, they are multiple ways to react (react on mouse down, react on mouse up, react on mouse down+up) which all have valid uses.

Typically for context menus it is more standard to react on mouse up which is what the BeginContextItemXXX functions do and which you can replicate with IsItemHovered() && IsMouseReleased(1)

If you actually need to detect a down+up sequence you’d need to use the full buttom behavior which you can use with eg InvisibleButton() with flags + AddImage().

**Comment by keszegh** [comment]
yes, i want a full down+up sequence, exactly as a normal 'click' works (i.e. when one does if one writes "if imgui:imageButton(...) then ..." (note that i use a lua/gideros implementation, from https://github.com/MultiPain/Gideros_ImGui but that does not matter much).  i don't know yet how "InvisibleButton() with flags + AddImage()" would work, sounds exactly like 'reinventing the wheel' though. however, if there is no other solution, then i'd appreciate a bit more details (which will surely save me hours of trial and error). 
overall i find it a bit surprising that in imgui there is no simple way to right-click on a button but maybe that's an unusual trick and only i like it to make my menus more compact by adding two behaviours to one button. 

thanks for the fast response.

**Comment by ocornut** [comment]
A majority of use of right-click are for context menu which have their own semantic.

I’ll let you look at the prototype and flags of InvisibleButton() or ButtonEx() you can select button mask - either function depending on the look you need (framed or not), and how you can use GetWindowDrawList()->AddImage() to render an image.

(ImageButton() is not adequate for legacy reason)

**Comment by keszegh** [comment]
btw i need to right-click similarly on a checkbox, that can be workarounded similarly?

**Comment by ocornut** [comment]
Not without making your own checkbox helper if you absolutely need down-up event (rather than just down). What would right click on a checkbox do?

**Comment by ocornut** [comment]
For neither you have answered what you want to do with the right-click. The correct answer almost always depends on what you intent to do with that right-click.

**Comment by keszegh** [comment]
in the first case i want to register a right-click on an imagebutton which would do something else as a left-click (what it does in my animation app is probably irrevelant). so the response is not related to imgui.
in the second case a right-click on a checkbox would relate to imgui. in one case it would set the checkbox in a third state (layer on/off state is normal and right-click would put it into a 'solo' state where instead of a tick an 'S' should be there. in other case it would split the checkbox into two checkboxes, going into 'advanced mode' (one checkbox affecting a property of a layer in the main window of my drawing app and the extra checkbox affecting same property in projection window, but these things are outside imgui - in nonadvanced mode the two behaviours are not separated so one checkbox is enough).

**Comment by keszegh** [comment]
for now i've decided to make do without right-click behaviours, it's perhaps better to align with user expectations anyway.
